applications/ShallowWaterApplication/custom_conditions/wave_condition.cpp

```diff
@@ -102,9 +104,9 @@ void WaveCondition<TNumNodes>::GetValuesVector(Vector& rValues, int Step) const
     IndexType counter = 0;
     for (IndexType i = 0; i < TNumNodes; i++)
     {
-        rValues[counter++] = r_geom[i].FastGetSolutionStepValue(VELOCITY_X, Step);
-        rValues[counter++] = r_geom[i].FastGetSolutionStepValue(VELOCITY_Y, Step);
-        rValues[counter++] = r_geom[i].FastGetSolutionStepValue(HEIGHT, Step);
+        rValues[counter++] = r_geom[i].FastGetSolutionStepValue(this->GetUnknownComponent(0), Step);
```

Comment:
Unfortunately, this is more "general", but less efficient. You can consider a static array, or something else in order to improve efficiency

Reply:
I've tested the efficiency and the cost is acceptable.

Reply:
How can I define the static array in derived classes? Do you have an example?

Reply:
You mean that the method is virtual ?, I did not see that, sorry

Reply:
> I've tested the efficiency and the cost is acceptable.

![imagen](https://user-images.githubusercontent.com/19991680/134360168-90544abe-1335-4847-a451-f720737a5260.png)